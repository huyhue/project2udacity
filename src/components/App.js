import { Route, Routes, Navigate } from 'react-router-dom';
import LoginPage from './LoginPage';
import Page404 from './Page404';
import { handleInitData } from '../actions/shared';
import LoadingBar from "react-redux-loading-bar";
import { useEffect, Fragment } from 'react';
import { connect } from "react-redux";
import Dashboard from "./Dashboard";
import NavigateBar from './NavigateBar';
import ChooseAnswerPage from "./ChooseAnswerPage";
import CreateNewQuestion from './CreateNewQuestion';
import Topboard from "./Topboard";

function App({ dispatch, loggedUser }) {

  useEffect(() => {
    dispatch(handleInitData());
  });

  const redirection = window.location.href.toString().split(window.location.host)[1];

  return (
    <Fragment>
      <LoadingBar />
      <div className="container">
      {loggedUser && <NavigateBar />}
          <Routes>
            <Route path="/login" exact element={<LoginPage />} />
            <Route path="*" element={loggedUser ? <Page404 /> : <Navigate to={`/login?redirectTo=${redirection}`}/>} />
            <Route path="/" exact element={loggedUser ? <Dashboard /> : <Navigate to={`/login?redirectTo=${redirection}`}/>} />
            <Route path="/add" element={loggedUser ? <CreateNewQuestion /> : <Navigate to={`/login?redirectTo=${redirection}`}/>} />
            <Route path="/questions/:id" element={loggedUser ? <ChooseAnswerPage /> : <Navigate to={`/login?redirectTo=${redirection}`}/>} />
            <Route path="/leaderboard" element={loggedUser ? <Topboard /> : <Navigate to={`/login?redirectTo=${redirection}`}/>} />
          </Routes>
      </div>
    </Fragment>
  );
}

const mapStateToProps = ({authedUser}) => ({
  loggedUser: !!authedUser,
});

export default connect(mapStateToProps)(App);
