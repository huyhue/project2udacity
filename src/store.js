import reducers from './reducers';
import middleware from "./middlewares";
import { legacy_createStore as createStore } from "redux";

const store = createStore(reducers, middleware);

export default store;
