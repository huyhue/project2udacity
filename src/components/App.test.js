import { render } from "@testing-library/react";
import '@testing-library/jest-dom'
import React from "react";
import {Provider} from "react-redux";
import store from "../store";
import {BrowserRouter} from "react-router-dom";
import App from  "./App";

describe("Root page", () => {
    it("should render screen", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App />
                </BrowserRouter>
            </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });

    it("should show login card if user is not log in yet", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App />
                </BrowserRouter>
            </Provider>
        );

        const quickLogin = component.getByTestId("quickLogin");
        expect(quickLogin).toBeInTheDocument();
    });
})