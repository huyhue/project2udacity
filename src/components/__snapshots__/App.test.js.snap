// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root page should render screen 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
      <div
        class="container"
      >
        <div>
          <div
            class="container"
          >
            <div
              class="vh-100 d-flex justify-content-center align-items-center row"
            >
              <div
                class="col-lg-6 col-md-8 col-12"
              >
                <div
                  class="border border-3 border-primary"
                />
                <div
                  class="shadow card"
                >
                  <div
                    class="card-body"
                  >
                    <div
                      class="mb-3 mt-md-4 text-center"
                    >
                      <h2
                        class="fw-bold mb-2 text-uppercase"
                      >
                        EMPLOYEE POLLS
                      </h2>
                      <p
                        class=" mb-3"
                      >
                        Please enter your ID and password
                      </p>
                      <strong>
                        ***Quick login***
                      </strong>
                      <div
                        class="align-items-center"
                        style="margin: 0px auto;"
                      >
                        <select
                          class="mb-3 mt-2 text-center form-select"
                          data-testid="quickLogin"
                        >
                          <option
                            value=""
                          >
                            Select user
                          </option>
                          <option
                            value="sarahedo"
                          >
                            sarahedo
                          </option>
                          <option
                            value="tylermcginnis"
                          >
                            tylermcginnis
                          </option>
                          <option
                            value="zoshikanlu"
                          >
                            zoshikanlu
                          </option>
                          <option
                            value="mtsamis"
                          >
                            mtsamis
                          </option>
                        </select>
                      </div>
                      <div
                        class="mb-3"
                      >
                        <form
                          class=""
                        >
                          <div
                            class="mb-3"
                          >
                            <label
                              class="text-center form-label"
                              for="formBasicText"
                            >
                              Your ID
                            </label>
                            <input
                              class="form-control"
                              data-testid="username"
                              id="formBasicText"
                              placeholder="ID"
                              type="text"
                              value=""
                            />
                          </div>
                          <div
                            class="mb-3"
                          >
                            <label
                              class="form-label"
                              for="formBasicPassword"
                            >
                              Password
                            </label>
                            <input
                              class="form-control"
                              data-testid="password"
                              id="formBasicPassword"
                              placeholder="Password"
                              type="password"
                              value=""
                            />
                          </div>
                          <div
                            class="d-grid"
                          >
                            <button
                              class="btn btn-primary"
                              data-testid="submit"
                              type="submit"
                            >
                              Login
                            </button>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div />
    <div
      class="container"
    >
      <div>
        <div
          class="container"
        >
          <div
            class="vh-100 d-flex justify-content-center align-items-center row"
          >
            <div
              class="col-lg-6 col-md-8 col-12"
            >
              <div
                class="border border-3 border-primary"
              />
              <div
                class="shadow card"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="mb-3 mt-md-4 text-center"
                  >
                    <h2
                      class="fw-bold mb-2 text-uppercase"
                    >
                      EMPLOYEE POLLS
                    </h2>
                    <p
                      class=" mb-3"
                    >
                      Please enter your ID and password
                    </p>
                    <strong>
                      ***Quick login***
                    </strong>
                    <div
                      class="align-items-center"
                      style="margin: 0px auto;"
                    >
                      <select
                        class="mb-3 mt-2 text-center form-select"
                        data-testid="quickLogin"
                      >
                        <option
                          value=""
                        >
                          Select user
                        </option>
                        <option
                          value="sarahedo"
                        >
                          sarahedo
                        </option>
                        <option
                          value="tylermcginnis"
                        >
                          tylermcginnis
                        </option>
                        <option
                          value="zoshikanlu"
                        >
                          zoshikanlu
                        </option>
                        <option
                          value="mtsamis"
                        >
                          mtsamis
                        </option>
                      </select>
                    </div>
                    <div
                      class="mb-3"
                    >
                      <form
                        class=""
                      >
                        <div
                          class="mb-3"
                        >
                          <label
                            class="text-center form-label"
                            for="formBasicText"
                          >
                            Your ID
                          </label>
                          <input
                            class="form-control"
                            data-testid="username"
                            id="formBasicText"
                            placeholder="ID"
                            type="text"
                            value=""
                          />
                        </div>
                        <div
                          class="mb-3"
                        >
                          <label
                            class="form-label"
                            for="formBasicPassword"
                          >
                            Password
                          </label>
                          <input
                            class="form-control"
                            data-testid="password"
                            id="formBasicPassword"
                            placeholder="Password"
                            type="password"
                            value=""
                          />
                        </div>
                        <div
                          class="d-grid"
                        >
                          <button
                            class="btn btn-primary"
                            data-testid="submit"
                            type="submit"
                          >
                            Login
                          </button>
                        </div>
                      </form>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
